advocacy_docs/pg_extensions/spl_check/using/tracer.mdx

```diff
@@ -0,0 +1,115 @@
+---
+title: Tracer
+---
+EDB SPL Check provides a tracing option where you can see notices on start or end functions (terse and default verbosity) and start or end statements (verbose verbosity). For default and verbose verbosity, this option displays the content of function arguments. The content of related variables diplay when verbosity is verbose. For example: 
+
+```sql
+postgres=# do $$ begin perform fx(10,null, 'now', e'stěhule'); end; $$;
+NOTICE:  #0 ->> start of inline_code_block (Oid=0)
+NOTICE:  #2   ->> start of function fx(integer,integer,date,text) (Oid=16405)
+NOTICE:  #2        call by inline_code_block line 1 at PERFORM
+NOTICE:  #2       "a" => '10', "b" => null, "c" => '2020-08-03', "d" => 'stěhule'
+NOTICE:  #4     ->> start of function fx(integer) (Oid=16404)
+NOTICE:  #4          call by fx(integer,integer,date,text) line 1 at PERFORM
+NOTICE:  #4         "a" => '10'
+NOTICE:  #4     <<- end of function fx (elapsed time=0.098 ms)
+NOTICE:  #2   <<- end of function fx (elapsed time=0.399 ms)
+NOTICE:  #0 <<- end of block (elapsed time=0.754 ms)
+```
+
+The number that follows `#` is the execution frame counter. This number is related to the depth of the error context stack. It allows you to pair the start and end of a function. 
+
+## Using the tracer
+
+To enable tracing, set `spl_check.tracer` to `on`. 
+
+!!!Warning 
+   Enabling tracing can have a significant impact on performance. To prevent performance issues, you can set a level for outputs used by the tracer with `spl_check.tracer_errlevel`. (The default is `notice`.) You can limit the output content according to the length specified with the `spl_check.tracer_variable_max_length` configuration variable. 
+
+As a security safeguard, we recommend that the use of the tracer be enabled by a superuser by setting `spl_check.enable_tracer` to `on` or `spl_check.enable_tracer` to `on` in `postgresql.conf`. Because the tracer shows the content of EDB SPL Check's variables, some sensitive security information can display for an unprivileged user. 
+
+We also recommend that you load the extension `spl_check`. You can execute some `spl_check` functions explicitly with `load 'spl_check';`. You can use the configuration's `shared_preload_libraries`, `local_preload_libraries`, or `session_preload_libraries` option. 
+
```

Comment:
Check edit (deleted "or"). Last sentence doesn't feel connected to the rest of the paragraph. What can you use those options for?

Reply:
I agree; this was unclear. Changing to: "You can also set the configuration options in `shared_preload_libraries`, `local_preload_libraries`, or `session_preload_libraries` options to load `spl_check`. "